src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -662,24 +790,70 @@ private static FormulaValue ModInternal(double arg0, double arg1, IRContext irCo
             return new NumberValue(irContext, result);
         }
 
+        public static FormulaValue ModDecimal(IRContext irContext, DecimalValue arg0dv, DecimalValue arg1dv)
+        {
+            decimal arg0 = arg0dv.Value;
+            decimal arg1 = arg1dv.Value;
+            decimal q;
+
+            if (arg1 == 0m)
+            {
+                return CommonErrors.DivByZeroError(irContext);
+            }
+
+            // r = a – N × floor(a/b)
+            try
+            {
+                q = decimal.Floor(arg0 / arg1);

```

Comment:
>q = decimal.Floor(arg0 / arg1); [](http://example.com/codeflow?start=16&length=31)

decimal.MaxValue % 0.7m works in C#

Reply:
Good call.  I followed the float algorithm.  I'm going to put an issue in on this and we can revisit, both for decimal and float.  https://github.com/microsoft/Power-Fx/issues/1367